okta/okta.go

```diff
@@ -189,18 +189,60 @@ func (m *userManager) doReauthorize(accessToken, refreshToken string) error {
 	return catcher.Resolve()
 }
 
-// reauthorizeUser attempts to reauthorize the user, first with their current
-// access token. If that fails, it refreshes the tokens and attempts to
+// reauthorizeID attempts to reauthorize the user based on their ID token.
+// Reauthorization succeeds if the username derived from their ID token matches
+// the given username.
+func (m *userManager) reauthorizeID(username string, tokens *tokenResponse) error {
+	idToken, err := m.doValidateIDToken(tokens.IDToken, "")
+	// When we receive an ID token from a refresh grant (i.e. during
+	// reauthorization), the ID token does not contain a nonce. However, the
+	// Okta token verification library requires that a nonce always be specified
+	// for verifying ID tokens.
+	// We ignore this error if it occurs. More info can be found here:
+	// https://bitbucket.org/openid/connect/issues/1025/ambiguity-with-how-nonce-is-handled-on
+	if err != nil && !strings.Contains(strings.ToLower(err.Error()), "nonce") {
```

Comment:
paranoia, but we should check the complete error ("the `Nonce` was not able to be validated"?)

Reply:
Done.